libgalois/src/PropertyGraph.cpp

```diff
@@ -229,6 +406,76 @@ katana::PropertyGraph::Copy(
   return Make(rdg_dir(), opts);
 }
 
+katana::Result<void>
+katana::PropertyGraph::ConstructTypeIDs() {
+  uint64_t num_node_rows = static_cast<uint64_t>(node_properties()->num_rows());
+  if (num_node_rows == 0) {
+    if ((node_properties()->num_columns() != 0) && (num_nodes() != 0)) {
+      return KATANA_ERROR(
+          ErrorCode::AssertionFailed,
+          "execution failed because number of rows in node properties is 0 but "
+          "the number of node properties is {} and the number of nodes is {}",
+          node_properties()->num_rows(), num_nodes());
+    }
+  } else if (num_node_rows != num_nodes()) {
+    return KATANA_ERROR(
+        ErrorCode::AssertionFailed,
+        "execution failed because number of rows in node properties {} differs "
+        "from the number of nodes {}",
+        node_properties()->num_rows(), num_nodes());
+  }
+  uint64_t num_edge_rows = static_cast<uint64_t>(edge_properties()->num_rows());
+  if (num_edge_rows == 0) {
+    if ((edge_properties()->num_columns() != 0) && (num_edges() != 0)) {
+      return KATANA_ERROR(
+          ErrorCode::AssertionFailed,
+          "execution failed because number of rows in edge properties is 0 but "
+          "the number of edge properties is {} and the number of edges is {}",
+          edge_properties()->num_rows(), num_edges());
+    }
+  } else if (num_edge_rows != num_edges()) {
+    return KATANA_ERROR(
+        ErrorCode::AssertionFailed,
+        "execution failed because number of rows in edge properties {} differs "
+        "from the number of edges {}",
+        edge_properties()->num_rows(), num_edges());
+  }
```

Comment:
These are useful checks. But I think they should be in `Make`. Having property arrays of the wrong length is an issue independently of types.